src/main/java/domain/Attendance.java

```diff
@@ -0,0 +1,179 @@
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class Attendance {
+
+    private final Map<Crew, List<AttendanceTime>> attendance;
+
+    public Attendance(Map<String, List<LocalDateTime>> attendance, LocalDate nowDate) {
+        this.attendance = new HashMap<>();
+        for (String name : attendance.keySet()) {
+            this.attendance.put(new Crew(name), attendance.get(name)
+                    .stream()
+                    .map(AttendanceTime::new)
+                    .collect(Collectors.toList()));
+        }
+        initializeAttendances(attendance, nowDate);
+    }
+
+    private void initializeAttendances(Map<String, List<LocalDateTime>> attendance, LocalDate nowDate) {
+        LocalDate startDate = LocalDate.of(2024, 12, 1);
+        LocalDate endDate = nowDate;
+        if (endDate.isAfter(LocalDate.of(2024, 12, 31))) {
+            endDate = LocalDate.of(2025, 1, 1);
+        }
+
+        for (String name : attendance.keySet()) {
+            initializeAttendance(name, startDate, endDate);
+        }
+    }
+
+    private void initializeAttendance(String name, LocalDate startDate, LocalDate endDate) {
+        List<AttendanceTime> attendanceTimes = this.attendance.get(findCrew(name));
+        Set<LocalDate> attendanceDates = attendanceTimes.stream()
+                .map(attendanceTime -> attendanceTime.getAttendanceDateTime().toLocalDate())
+                .collect(Collectors.toSet());
+
+        for (LocalDate date = startDate; date.isBefore(endDate); date = date.plusDays(1)) {
+            attendanceTimes = addUnattended(attendanceDates, date, attendanceTimes);
+        }
+        this.attendance.put(findCrew(name), attendanceTimes);
+    }
+
+    private Crew findCrew(String name) {
+        return attendance.keySet().stream()
+                .filter(crew -> crew.getName().equals(name))
+                .findAny()
+                .orElseThrow(() -> new IllegalArgumentException("[ERROR] 존재하지 않는 크루입니다."));
+    }
+
+    private List<AttendanceTime> addUnattended(Set<LocalDate> attendanceDates, LocalDate date,
+                                               List<AttendanceTime> attendanceTimes) {
+        if (!(attendanceDates.contains(date) || isClosed(date))) {
+            List<AttendanceTime> updatedAttendanceTimes = new ArrayList<>(attendanceTimes);
+            updatedAttendanceTimes.add(new AttendanceTime(date, AttendanceStatus.UNATTEND));
+            return updatedAttendanceTimes;
+        }
+        return attendanceTimes;
+    }
+
+    public boolean isClosed(LocalDate date) {
+        return date.getDayOfWeek() == DayOfWeek.SATURDAY || date.getDayOfWeek() == DayOfWeek.SUNDAY || date.isEqual(
+                LocalDate.of(2024, 12, 25));
+    }
```

Comment:
```suggestion
    public boolean isClosed(LocalDate date) {
        return date.getDayOfWeek() == DayOfWeek.SATURDAY
            || date.getDayOfWeek() == DayOfWeek.SUNDAY
            || date.isEqual(LocalDate.of(2024, 12, 25));
    }
```

가독성에 신경을 많이 써주셔서 남겨보는 리뷰입니다.
개인적으로 여러 조건을 검사할 때는 이렇게 줄바꿈을 해주면 이해하기 편하더라구요.
디버깅 하기도 좋구요 😄